tests/cpp/test_parallel_executor.cpp

```diff
@@ -0,0 +1,25 @@
+#include "taichi/common/testing.h"
+#include "taichi/program/async_engine.h"
+
+TLANG_NAMESPACE_BEGIN
+
+TI_TEST("parallel_executor") {
+  SECTION("create_and_destruct") {
+    ParallelExecutor exec(10);
+  }
+  SECTION("parallel_print") {
+    int N = 100;
+    std::vector<int> buffer(N, 0);
+    {
+      ParallelExecutor exec(10);
+      for (int i = 0; i < N; i++) {
+        exec.enqueue([i = i, &buffer]() { buffer[i] = i + 1; });
```

Comment:
Is `i = i` to explicitly capture by copying here? Is there a clearer way to write this?

I don't get the point if capture by reference can cause any problems...

Reply:
If we capture by reference, then when we actually use `i` its value will have been changed (or even released from stack...)